modules/class-simplechart-wp-cli.php

```diff
@@ -0,0 +1,246 @@
+			return null;
+		} elseif ( 200 !== $code = wp_remote_retrieve_response_code( $response ) ) {
+			WP_CLI::warning( sprintf( __( 'Post %s from site %s returned error code %s', 'simplechart' ), $post, $site, $code ) );
+			return null;
+		}
+
+		$post_object = json_decode( wp_remote_retrieve_body( $response ), true );
+
+		// make sure it's the right post type
+		if ( $this->_post_type !== $post_object['type'] ) {
+			WP_CLI::warning( sprintf( __( "Post %s from site %s does not have post_type '%s'", 'simplechart' ), $post, $site, $this->_post_type ) );
+			return null;
+		}
+
+		WP_CLI::success( sprintf( __( 'Retrieved post %s from site %s', 'simplechart' ), $post, $site ) );
+		return $post_object;
+	}
+
+	/**
+	 * Create a post in Simplechart post type using WPCOM REST API data
+	 * @param array $post_object API data for the post
+	 * @return array Post object transformed for our data loader
+	 */
+	private function _transform_from_wpcom_api( $post_object ) {
+		// basic post info
+		$post_data = array(
+			'post_name' => $post_object['slug'],
+			'post_title' => $post_object['title'],
+			'post_status' => $post_object['status'],
+			'post_type' => $this->_post_type,
+			'post_date_gmt' => date( 'Y-m-d H:i:s', strtotime( $post_object['date'] ) ),
+		);
+
+		// author ID if we have one
+		if ( $this->_author ) {
+			$post_data['post_author'] = $this->_author;
+		}
+
+		// setup post metadata
+		$post_meta = array(
+			'_src_guid' => $post_data['guid'],
+		);
+		if ( ! empty( $post_object['metadata'] ) && is_array( $post_object['metadata'] ) ) {
+			foreach ( $post_object['metadata']  as $meta ) {
+				if ( in_array( $meta['key'], $this->_meta_keys, true ) ) {
+					$post_meta[ $meta['key'] ] = $meta['value'];
+				}
+			}
+		}
+
+		// featured image URL
+		$featured_image = ! empty( $post_object['featured_image'] ) ? $post_object['featured_image'] : null;
+
+		return array(
+			'post_data' => $post_data,
+			'post_meta' => $post_meta,
+			'featured_image' => $featured_image,
+		);
+	}
+
+	/**
+	 * create new post for the chart, using already transformed data
+	 * @var array $data Post data for new chart
+	 *			array 'post_data' Arguments to pass to wp_insert_post()
+	 *			array 'post_meta' Key-value pairs to set as post meta
+	 *			array 'featured_image' Optional array of url, width, height of image to import
+	 * @return bool Success or failure
+	 */
+	private function _load_post( $data ) {
+		if ( empty( $data['post_data'] ) ) {
+			WP_CLI::warning( __( 'Cannot insert post with no post_data', 'simplechart' ) );
+			return false;
+		}
+
+		// create the post
+		$post_id = wp_insert_post( $data['post_data'] );
+		if ( 0 === $post_id ) {
+			WP_CLI::warning( __( 'Failed to insert post', 'simplechart' ) );
+			return false;
+		}
+
+		// add the post metadata
+		foreach ( $data['post_meta'] as $key => $value ) {
+			update_post_meta( $post_id, $key, $value );
+		}
+
+		// attach the featured image
+		if ( $data['featured_image'] ) {
```

Comment:
same as above, we can be confident in the structure of `$data` since there's only one way it can be set and we control it